browser/src/UI/components/BufferScrollBar.tsx

```diff
@@ -76,13 +78,44 @@ export class BufferScrollBar extends React.PureComponent<IBufferScrollBarProps,
                 height: size,
                 backgroundColor: m.color,
                 width: "100%",
+                pointerEvents: "none",
             }
 
             return <div style={markerStyle} key={`${this.props.windowId}_${m.color}_${m.line}`} />
         })
 
+        let scrollBarTop: number
```

Comment:
@jordan-arenstein thanks for the PR 😍, one slightly cosmetic-ish thing could the functions in the render be methods on the class and the `scrollBarTop` be in state so that its more explicit that the value is being changed?

Reply:
Thanks for the review @Akin909. I'm new to Flux and React, so I'm not 100% on the changes. I've committed, let me know if it's okay 😄 👍

Reply:
@jordan-arenstein no worries the changes I'm suggesting are that `setLine` `beginScroll` and `trackScroll` can be used as methods for the react component rather than re-created every time the component re-renders which is currently the case e.g.

```ts
interface IState {
	scrollBarTop: number
}
export class BufferScrollBar extends React.PureComponent<IBufferScrollBarProps, IState>
	public state = {
		scrollBarTop: null,
	}
	componentDidMount(){
		e.preventDefault()
        // offsetY is definitely on the scrollbar in the beginning of the click
         scrollBarTop = e.nativeEvent.clientY - e.nativeEvent.offsetY
         setLine(e.nativeEvent.clientY)
         document.addEventListener("mousemove", this.trackScroll, true)
         document.addEventListener("mouseup", this.endScroll, true)
	}
	
	componentWillUnmount() {
		document.removeEventListener("mousemove", this.trackScroll, true)
        document.removeEventListener("mouseup", this.endScroll, true)
	}

	// Above the render function
	public setLine = (y: number) => {
		const lineFraction = Math.min(Math.max((y - this.state.scrollBarTop) / this.props.height, 0), 1)
            const newLine = Math.ceil(
                editorManager.activeEditor.activeBuffer.lineCount * lineFraction,
            )
            editorManager.activeEditor.activeBuffer.setCursorPosition(newLine, 0)
	}

	public beginScroll = (e: React.MouseEvent<HTMLDivElement>) => {
            e.preventDefault()
            // offsetY is definitely on the scrollbar in the beginning of the click
            this.setState({ scrollBarTop: e.nativeEvent.clientY - e.nativeEvent.offsetY })
            setLine(e.nativeEvent.clientY)
            document.addEventListener("mousemove", trackScroll, true)
            document.addEventListener("mouseup", endScroll, true)
        }
```

Currently its implicit that the scrollBarTop bit of information is being changed, and is a dependency for other functions react aims to make state explicit and provides an api for changing which is `setState` by setting state the component updates correctly based on the new state and the state changes can be accessed inside of react `lifeCycleMethods` like `componentDidUpdate`.

Currently the functionality should be largely unchanged by switching to this but it gives us a good jumping of point for any future changes, lemme know if any of that doesn't make sense

Reply:
Okay I was on the right track. I've pushed my changes over an hour ago, but they aren't showing here?? They're visible on my fork though. I'm trying to figure it out.